src/components/sdk/MeetingControls/AudioInputVFControl.tsx

```diff
@@ -0,0 +1,140 @@
+  const { muted, toggleMute } = useToggleLocalMute();
+  const [device, setDevice] = useState<Device | AudioTransformDevice | null>(meetingManager.selectedAudioInputTransformDevice);
+  const { supportsVoiceFocus, addVoiceFocus } = useVoiceFocus();
+  const audioInputConfig: DeviceConfig = {
+    additionalDevices: true,
+  };
+  const { devices, selectedDevice } = useAudioInputs(audioInputConfig);
+
+  const dropdownOptions: PopOverItemProps[] = devices.map((device) => ({
+    children: <span>{device.label}</span>,
+    checked: isOptionActive(selectedDevice, device.deviceId),
+    onClick: (): Promise<void> =>
+      meetingManager.selectAudioInputDevice(device.deviceId),
+  }));
+
+  useEffect(() => {
+    if (device instanceof VoiceFocusTransformDevice) {
+      setEnableVoiceFocus(true);
+    } else {
+      setEnableVoiceFocus(false);
+    }
+  }, [device]);
+
+  useEffect(() => {
+    meetingManager.subscribeToSelectedAudioInputTransformDevice(setDevice);
+
+    return (): void => {
+      meetingManager.unsubscribeFromSelectedAudioInputTransformDevice(setDevice);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (supportsVoiceFocus) {
+      const vfOption: PopOverItemProps = {
+        children: <><PopOverSeparator />{isLoading && <Spinner />} {enableVoiceFocus ? <span>Disable voice focus</span> : <span>Enable voice focus</span>}</>,
+        checked: enableVoiceFocus,
+        disabled: isLoading,
+        onClick: () => {
+          setVoiceFocusOn(current => !current);
+        },
+      };
+      dropdownOptions?.push(vfOption);
+    }
+    dropdownWithVFOptions.current = dropdownOptions;
+  }, [supportsVoiceFocus, enableVoiceFocus, voiceFocusOn, isLoading, dropdownOptions]);
+
+  useEffect(() => {
+    async function onVFCheckboxChange() {
+      let current;
+      setIsLoading(true);
+      if (voiceFocusOn) {
+        if (typeof (device) === 'string') {
+          const currentDevice = audioInputSelectionToDevice(device);
+          current = await addVoiceFocus(currentDevice);
+        }
+      } else {
+        if (device instanceof VoiceFocusTransformDevice) {
+          current = device.getInnerDevice();
+        }
+      }
+      if (current) {
+        setDevice(current)
+        meetingManager.selectAudioInputDevice(current);
+      }
+      setIsLoading(false);
+    }
+
+    onVFCheckboxChange();
+  }, [voiceFocusOn]);
+
+  useEffect(() => {
+    async function onChoseNewDevice() {
+      if (voiceFocusOn) {
+        setIsLoading(true);
+        const VFdevice = meetingManager.selectedAudioInputTransformDevice;
+        if (typeof (device) === 'string' && VFdevice instanceof VoiceFocusTransformDevice) {
+          const currentDevice = audioInputSelectionToDevice(device);
+          const current = await VFdevice.chooseNewInnerDevice(currentDevice);
+          setDevice(current);
+        }
+        setIsLoading(false);
+      } else {
+        meetingManager.selectAudioInputDevice(device);
```

Comment:
I think we should `await` these `meetingManager.selectAudioInputDevice` calls since it returns a promise.